dask/dataframe/core.py

```diff
@@ -3202,7 +3342,7 @@ def apply(self, func, convert_dtype=True, meta=no_default, args=(), **kwds):
             warnings.warn(meta_warning(meta))
 
         return map_partitions(
-            M.apply, self, func, convert_dtype, args, meta=meta, **kwds
+            M.apply, self, func, convert_dtype, args, meta=meta, preserve_partitions=True, **kwds
```

Comment:
The `preserve_partitions` flag is a hint that we are transforming partition data in a way that is 1:1, granularity wise, with the previous data.  So we know we do can keep the parition_sizes metadata.  This is not always the case when mapping.